tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,891 @@
+        cls.columns = {
+            Estimates.event_date: 'event_date',
+            Estimates.fiscal_quarter: 'fiscal_quarter',
+            Estimates.fiscal_year: 'fiscal_year',
+            Estimates.estimate: 'estimate'
+        }
+        cls.loader = cls.make_loader(cls.events, {column.name: val for
+                                                  column, val in
+                                                  cls.columns.items()})
+        cls.ASSET_FINDER_EQUITY_SIDS = list(
+            cls.events[SID_FIELD_NAME].unique()
+        )
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(WithEstimates, cls).init_class_fixtures()
+
+
+class WithWrongNumQuarters(WithEstimates):
+    """
+    ZiplineTestCase mixin providing cls.events as a class level fixture and
+    defining a test for all inheritors to use.
+
+    Attributes
+    ----------
+    events : pd.DataFrame
+        A simple DataFrame with columns needed for estimates and a single sid
+        and no other data.
+
+    Tests
+    ------
+    test_wrong_num_quarters_passed()
+        Tests that loading with an incorrect quarter number raises an error.
+    """
+    events = pd.DataFrame({SID_FIELD_NAME: 0},
+                          columns=[SID_FIELD_NAME,
+                                   TS_FIELD_NAME,
+                                   EVENT_DATE_FIELD_NAME,
+                                   FISCAL_QUARTER_FIELD_NAME,
+                                   FISCAL_YEAR_FIELD_NAME,
+                                   'estimate'],
+                          index=[0])
+
+    def test_wrong_num_quarters_passed(self):
+        with self.assertRaises(ValueError):
+            dataset = QuartersEstimates(-1)
+            engine = SimplePipelineEngine(
+                lambda x: self.loader,
+                self.trading_days,
+                self.asset_finder,
+            )
+
+            engine.run_pipeline(
+                Pipeline({c.name: c.latest for c in dataset.columns}),
+                start_date=self.trading_days[0],
+                end_date=self.trading_days[-1],
+            )
+
+
+class PreviousWithWrongNumQuarters(WithWrongNumQuarters,
+                                   ZiplineTestCase):
+    """
+    Tests that previous quarter loader correctly breaks if an incorrect
+    number of quarters is passed.
+    """
+    @classmethod
+    def make_loader(cls, events, columns):
+        return PreviousQuartersEstimatesLoader(events, columns)
+
+
+class NextWithWrongNumQuarters(WithWrongNumQuarters,
+                               ZiplineTestCase):
+    """
+    Tests that next quarter loader correctly breaks if an incorrect
+    number of quarters is passed.
+    """
+    @classmethod
+    def make_loader(cls, events, columns):
+        return NextQuartersEstimatesLoader(events, columns)
+
+
+class WithEstimatesT0(WithEstimates):
```

Comment:
Time 0 - the immediate next/previous quarter on which all other calculations should be based.